### PR TITLE
Appends follow-up UI fixes

### PR DESCRIPTION
**Pull request by iethree**
### Description

A couple little things came up during today's demo:

1) we should auto-select the only model to append if there's only 1
only-one

2) there was a bug where[ if you did certain things in the ui](https://www.loom.com/share/4f65e82e1d054aebbc58716d3a31702a) (like create a new question), we would re-trigger an error modal.

### Tests
- [x] tests have been added

### CODE REVIEW

**Comment by mazameli** [comment]
> we should auto-select the only model to append if there's only 1

IMO, if there are _any_ to append to, we should just auto-select one of them, since even dumbly selecting the first one alphabetically has a higher probability of being the one the user intended (>0% chance of being right) than the probability that no selection is the correct one (exactly 0%, haha). If it's easy to be slightly smarter (like auto selecting the one with the most recent last_edited_at date or something), so much the better.

**Comment by iethree** [comment]
> > we should auto-select the only model to append if there's only 1
> 
> IMO, if there are _any_ to append to, we should just auto-select one of them, since even dumbly selecting the first one alphabetically has a higher probability of being the one the user intended (>0% chance of being right) than the probability that no selection is the correct one (exactly 0%, haha). If it's easy to be slightly smarter (like auto selecting the one with the most recent last_edited_at date or something), so much the better.

ok, now this will auto-select the most recently update model (which will also be the only model if there's just one)

File: frontend/src/metabase/collections/components/ModelUploadModal.tsx [hunk 1]
```diff
@@ -47,6 +49,18 @@ export function ModelUploadModal({
     [models.data],
   );
 
+  useEffect(
+    function setDefaultTableId() {
```
Comment (iethree) [inline frontend/src/metabase/collections/components/ModelUploadModal.tsx:53]: I named this function especially for @sloansparger

**Review by sloansparger — APPROVED** [review]
lgtm!

**Pull request merged**